rust/src/log.rs

```diff
@@ -202,6 +202,22 @@ pub fn sc_log_message(level: Level,
     return 0;
 }
 
+// Convert a Vec into a CString by first stripping NUL bytes.
+pub fn vec_to_safe_cstring(val: Vec<u8>) -> CString {
```

Comment:
As mentioned, seems out of place is the module. As it is only to support the specialized base64decode function, should probably live alongside it.